port/raspberrypi/rp2xxx/src/hal/pio/assembler/encoder.zig

```diff
@@ -624,22 +681,22 @@ const expect = std.testing.expect;
 const expectEqual = std.testing.expectEqual;
 const expectEqualStrings = std.testing.expectEqualStrings;
 
-fn encode_bounded_output_impl(source: []const u8, diags: *?assembler.Diagnostics) !Encoder(.{}).Output {
-    const tokens = try tokenizer.tokenize(source, diags, .{});
-    var encoder = Encoder(.{}).init(tokens.slice());
+fn encode_bounded_output_impl(comptime chip: Chip, source: []const u8, diags: *?assembler.Diagnostics) !Encoder(chip, .{}).Output {
+    const tokens = try tokenizer.tokenize(chip, source, diags, .{});
+    var encoder = Encoder(chip, .{}).init(tokens.slice());
     return try encoder.encode_output(diags);
 }
 
-fn encode_bounded_output(source: []const u8) !Encoder(.{}).Output {
+fn encode_bounded_output(comptime chip: Chip, source: []const u8) !Encoder(chip, .{}).Output {
```

Comment:
I would like to run this for both chips, but the return type depends on the type.